packages/trader/src/App/Components/Elements/VideoPlayer/video-player.tsx

```diff
@@ -32,30 +32,36 @@ const VideoPlayer = ({ src, is_mobile, data_testid }: TVideoPlayerProps) => {
     const progress_bar_ref = React.useRef<HTMLDivElement>(null);
     const progress_dot_ref = React.useRef<HTMLSpanElement>(null);
     const animation_ref = React.useRef(0);
-    const timer_ref = React.useRef<ReturnType<typeof setTimeout>>();
+    const play_on_rewind_timer_ref = React.useRef<ReturnType<typeof setTimeout>>();
+    const replay_timer_ref = React.useRef<ReturnType<typeof setTimeout>>();
     const is_dragging = React.useRef(false);
 
-    const togglePlay = (
-        e:
-            | React.MouseEvent<HTMLDivElement | HTMLButtonElement>
-            | React.KeyboardEvent<HTMLDivElement | HTMLButtonElement>
-    ) => {
-        e.stopPropagation();
+    const togglePlay = React.useCallback(
+        (
+            e:
+                | React.MouseEvent<HTMLDivElement | HTMLButtonElement>
+                | React.KeyboardEvent<HTMLDivElement | HTMLButtonElement>
+        ) => {
+            e.stopPropagation();
 
-        if (!video_ref.current || !progress_bar_filled_ref.current) return;
+            if (!video_ref.current || !progress_bar_filled_ref.current) return;
 
-        if (is_playing) {
-            video_ref.current.pause();
-            setIsAnimated(false);
-            cancelAnimationFrame(animation_ref.current);
-        } else {
-            video_ref.current.play();
-            setIsAnimated(true);
-        }
+            if (is_playing) {
+                video_ref.current.pause();
+                setIsAnimated(false);
+                cancelAnimationFrame(animation_ref.current);
+            } else {
+                video_ref.current.play();
+                replay_timer_ref.current = setTimeout(() => {
+                    setIsAnimated(true);
+                }, 100);
```

Comment:
this timeout is needed to avoid weird animation rewinding to the left when replay overlay is clicked to restart playing. Result is in the recording (observe that upon replay, it simply starts playing from the left side without extra animation):

https://github.com/kate-deriv/deriv-app/assets/103177211/62619c5b-cbf4-42dc-8c31-60c16c892af9